checkov/common/bridgecrew/platform_integration.py

```diff
@@ -197,17 +245,46 @@ def get_id_mapping(self) -> dict:
             self.get_checkov_mapping_metadata()
         return self.bc_id_mapping
 
+    def get_ckv_to_bc_id_mapping(self) -> dict:
+        if not self.ckv_to_bc_id_mapping:
+            self.get_checkov_mapping_metadata()
+        return self.ckv_to_bc_id_mapping
+
     def get_checkov_mapping_metadata(self) -> dict:
         try:
             request = http.request("GET", self.guidelines_api_url)
             response = json.loads(request.data.decode("utf8"))
             self.guidelines = response["guidelines"]
             self.bc_id_mapping = response.get("idMapping")
+            self.ckv_to_bc_id_mapping = {ckv_id: bc_id for (bc_id, ckv_id) in self.bc_id_mapping.items()}
             logging.debug(f"Got checkov mappings from Bridgecrew BE")
         except Exception as e:
             logging.debug(f"Failed to get the guidelines from {self.guidelines_api_url}, error:\n{e}")
             return {}
 
+    def get_platform_fixes(self, scan_results):
```

Comment:
let's move fixes flow to a new python class.
1st python class will handle vscode integration
2ns python class will handle CI integration

wdyt?

Reply:
It's 99% independent of vscode. The only vscode specific logic is to skip the S3 stuff. Suppressions and fixes are just pure checkov features.

We should probably refactor all of these flows into new classes:

- dev API integrations (suppressions and fixes)
- old scanning logic (upload to S3)
- signup flow

And use the platform integration class just to hold the relevant attributes.

I'll try out refactoring just the dev API stuff to see how it looks. Maybe a "platform integration" class with abstract methods for setup, pre-scan, and post-scan logic? And the implementation classes just pull relevant state attributes from the platform_integration class.

Reply:
I used the check registry model to create a "platform integration registry". This made it trivial to convert the logic over.

It should work well to convert the S3 logic over as well, possible with more integration hooks (e.g., pre_scan). The only one I'm not sure of is the guidelines integration, because that one is a dependency for the runners. I am not sure what the best mechanism is to fetch guidelines and then make them available everywhere.